crucible-syntax/src/Lang/Crucible/Syntax/Concrete.hs

```diff
@@ -191,6 +191,10 @@ defaultParserHooks = ParserHooks empty empty
 data ParsedProgram ext = ParsedProgram
   { parsedProgGlobals :: Map GlobalName (Pair TypeRepr GlobalVar)
     -- ^ The parsed @defglobal@s.
+  , parsedProgExterns :: Map GlobalName (Pair TypeRepr GlobalVar)
```

Comment:
As I note in #1053, I'm not quite sure how I feel about the use of `Pair TypeRepr GlobalVar` here, as it really ought to be `Some GlobalVar` instead. On the other hand, `parsedProgGlobals` already uses `Pair TypeRepr GlobalVar`, and the types of `parsedProgGlobals` and `parsedProgExterns` really ought to be the same. I suppose we could fix #1053 now and change `Pair TypeRepr` to `Some` everywhere, but that would be an unforced API change.

Reply:
I agree that the `Some` approach is cleaner, and I'm also sympathetic to not wanting to change the API without a good reason.  That being said, I don't think there are many `crucible-syntax` users. @langston-barrett is the only other person I'm aware of who may be using it, so maybe they can chime in with whether this would be painful for them to accommodate?  Is there anyone else we should check with who might be impacted by changing the type?

I'm happy to go either way on this one.

Reply:
I went ahead and pushed a fix for #1053 as a separate commit. If folks don't want API churn, I can always revert that commit.

Reply:
> @langston-barrett is the only other person I'm aware of who may be using it, so maybe they can chime in with whether this would be painful for them to accommodate?

I'm not using it!

Reply:
Ah, good to know. In that case, I will be bold and change the use of `Pair TypeRepr` to `Some`, since I expect the disruption to be minimal.